pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java

```diff
@@ -68,10 +68,10 @@ public enum AggregationFunctionType {
   MODE("mode"),
 
   FIRSTWITHTIME("firstWithTime", null, SqlKind.OTHER_FUNCTION, SqlFunctionCategory.USER_DEFINED_FUNCTION,
-      OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.TIMESTAMP, SqlTypeFamily.CHARACTER)),
+      OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER)),
```

Comment:
This essentially is only a arg count checker as the v2 engine doesn't rely on the third arg for return type (uses ARG0 type inference)

Reply:
ok found a solution
```suggestion
      OperandTypes.or(
          OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.TIMESTAMP, SqlTypeFamily.CHARACTER)),
          OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.NUMERIC, SqlTypeFamily.CHARACTER))
      ),
```

Reply:
Nice, let me update this .

Reply:
We should re-visit all the ANY usage then.

Reply:
the third argument is inherited from v1 function, or maybe we should make it optional?

Reply:
Our internal code still check the third argument, so I guess let's not touch that for now.

Reply:
thank you for addressing this. :-D